lib/models/src/main/java/org/sagebionetworks/repo/model/drs/IdAndTypeParser.java

```diff
@@ -0,0 +1,82 @@
+package org.sagebionetworks.repo.model.drs;
+
+import org.apache.commons.lang3.StringUtils;
+import org.sagebionetworks.repo.model.entity.IdAndVersion;
+import org.sagebionetworks.util.ValidateArgument;
+
+/**
+ * Immutable representation of a DRS Object ID
+ */
+public class IdAndTypeParser {
+	public enum Type {
+		FILE_ENTITY,
+		FILE_HANDLE;
+	}
+
+	private static final String ENTITY_ID_PREFIX = "syn";
+	private static final String FILE_HANDLE_ID_PREFIX = "fh";
+
+	private Type type;
+	private IdAndVersion idAndVersion;
+
+	public IdAndTypeParser(Type type, IdAndVersion idAndVersion) {
```

Comment:
we can have another constructor(Type type, Long fileHandleid) . we can also remove type because with call of constructor we can find it out what type it is